Fixed direct-send RPC-API to accept list of UTXOs

amitx13 opened the pull request:
This PR fixes #1712 and addresses https://github.com/joinmarket-webui/jam/issues/772
Changes Made:

1. Updated the `direct-send` API to accept a list of `UTXOs` specified by the user.
2. Implemented backend logic to handle the selection of these `UTXOs` for the transaction.
3. Added necessary `validation checks` to ensure the specified UTXOs are part of the `user's wallet` and are `unfrozen`.

kristapsk commented on src/jmclient/taker_utils.py:
```diff
@@ -36,6 +36,7 @@ def get_utxo_scripts(wallet: BaseWallet, utxos: dict) -> list:
 
 def direct_send(wallet_service: WalletService,
                 mixdepth: int,
+                selected_utxos: List[str],
```
This should not be required, should be optional.

amitx13 commented:
Thank you for the feedback. I will ensure the `RPC-API` maintains `backwards compatibility` and update the `documentation` as well.

amitx13 commented:
Hey @kristapsk I am done with the requested changes. Please have a look 🙏

amitx13 commented:
Hey @kristapsk if you have a moment, could you please take a look at `docs/api/wallet-rpc.yaml` I tried everything from my side but still, `OpenAPI Diff` keeps failing. It will be a great help for `me` as well as for `JAM` and `JoinMarket` users

amitx13 commented:
Hey @kristapsk if you have a moment could you please `re-run` the workflows.

amitx13 commented:
Hey, I don't know if I am correct, but a friend of mine recommended creating a new PR from a new branch to fix this OpenAPI Diff issue.This way, the `OpenAPI Diff` action will not detect a breaking change since, in the history of that branch, `selected_utxos` was never required.

kristapsk commented:
> Hey, I don't know if I am correct, but I think creating a new PR from a new branch to fix this OpenAPI Diff issue.This way, the `OpenAPI Diff` action will not detect a breaking change since, in the history of that branch, `selected_utxos` was never required. New PR to address this issue: #1721

If it solves it, then there's bug in OpenAPI Diff action. Intent is for it to compare against master branch. But you could try!

amitx13 commented:
> > Hey, I don't know if I am correct, but I think creating a new PR from a new branch to fix this OpenAPI Diff issue.This way, the `OpenAPI Diff` action will not detect a breaking change since, in the history of that branch, `selected_utxos` was never required. New PR to address this issue: #1721
> 
> If it solves it, then there's bug in OpenAPI Diff action. Intent is for it to compare against master branch. But you could try!

Oh, okay. Then why is the `OpenAPI Diff` action always showing breaking changes? I introduced a parameter, marked it as optional, and updated the wallet-rpc.yaml just like you did with `txfee` and `tumbler_options`.

roshii commented:
I did not meant to approve, this was a comment only. My apologies.
